src/main/java/com/google/api/codegen/transformer/php/PhpGapicSurfaceTransformer.java

```diff
@@ -228,6 +255,117 @@ private ViewModel buildClientViewModel(GapicInterfaceContext context) {
+        .interfaceKey(context.getInterface().getFullName())
+        .apiMethods(generateRestMethodConfigViews(context))
+        .build();
+  }
+
+  private List<RestMethodConfigView> generateRestMethodConfigViews(GapicInterfaceContext context) {
+    List<RestMethodConfigView> configViews = new ArrayList<>();
+
+    for (MethodModel methodModel : context.getSupportedMethods()) {
+      GapicMethodContext methodContext = context.asDynamicMethodContext(methodModel);
+
+      if (!methodContext.getMethodConfig().isGrpcStreaming()) {
+        configViews.add(generateRestMethodConfigView(methodContext));
+      }
+    }
+
+    return configViews;
+  }
+
+  private RestMethodConfigView generateRestMethodConfigView(GapicMethodContext context) {
+    RestMethodConfigView.Builder restMethodConfig = RestMethodConfigView.newBuilder();
+    Method method = context.getMethod();
+    Map<String, String> httpMethodMap = new HashMap<>();
+    HttpRule httpRule = getHttpRule(method.getOptionFields());
+    String body = httpRule.getBody();
+
+    httpMethodMap.put("get", httpRule.getGet());
+    httpMethodMap.put("put", httpRule.getPut());
+    httpMethodMap.put("delete", httpRule.getDelete());
+    httpMethodMap.put("post", httpRule.getPost());
+    httpMethodMap.put("patch", httpRule.getPatch());
+
+    for (Map.Entry<String, String> entry : httpMethodMap.entrySet()) {
+      String uriTemplate = entry.getValue();
+
+      if (uriTemplate.isEmpty()) {
+        continue;
+      }
+
+      Set<String> templateVars = PathTemplate.create(uriTemplate).vars();
+
+      restMethodConfig.placeholders(generateRestPlaceholderConfigViews(context, templateVars));
+      restMethodConfig.hasPlaceholders(templateVars.size() > 0);
+      restMethodConfig.method(entry.getKey());
+      restMethodConfig.uriTemplate(uriTemplate);
+
+      break;
+    }
+
+    restMethodConfig.name(method.getSimpleName());
+    restMethodConfig.hasBody(!body.isEmpty());
+    restMethodConfig.body(body);
+
+    return restMethodConfig.build();
+  }
+
+  private List<RestPlaceholderConfigView> generateRestPlaceholderConfigViews(
+      GapicMethodContext context, Set<String> templateVars) {
+    List<RestPlaceholderConfigView> placeholderViews = new ArrayList<>(templateVars.size());
+
+    for (String var : templateVars) {
+      placeholderViews.add(generateRestPlaceholderConfigView(context, var));
+    }
+
+    return placeholderViews;
+  }
+
+  private RestPlaceholderConfigView generateRestPlaceholderConfigView(
+      GapicMethodContext context, String var) {
+    SurfaceNamer namer = context.getNamer();
+    MethodModel methodModel = context.getMethodConfig().getMethodModel();
+    RestPlaceholderConfigView.Builder placeholderView = RestPlaceholderConfigView.newBuilder();
+    String[] getters = var.split("\\.");
+    String fullName = context.getSurfaceInterfaceContext().getInterface().getFullName();
+
+    for (int i = 0; i < getters.length; i++) {
+      getters[i] =
```

Comment:
I think I would prefer adding to a new list instead of modifying the string array in place.